### ISSUE TITLE
UnicodeEncodeError: 'ascii' codec can't encode character u'\u201c' in position 18498: ordinal not in range(128)

### ISSUE THREAD

**Issue by anthony-estevez-zefr** [venth/aws-adfs#25]
Hello,

I'm getting this issue when I attempt to login.

aws-adfs version: 0.3.0
Mac version: 10.12.4
Python: 2.7.10

```
2017-03-31 14:20:10,683 [html_roles_fetcher html_roles_fetcher.py:fetch_html_encoded_roles] [90962-MainProcess] [140736347890624-MainThread] - DEBUG: Attempt to load authentication cookies into session failed. Re-authentication will be performed. The error:
2017-03-31 14:20:10,730 [connectionpool connectionpool.py:_new_conn] [90962-MainProcess] [140736347890624-MainThread] - DEBUG: Starting new HTTPS connection (1): SITE
2017-03-31 14:20:11,114 [connectionpool connectionpool.py:_make_request] [90962-MainProcess] [140736347890624-MainThread] - DEBUG: https://SITE:443 "POST /adfs/ls/IdpInitiatedSignOn.aspx?loginToRp=urn:amazon:webservices HTTP/1.1" 200 19245
Traceback (most recent call last):
  File "/usr/local/bin/aws-adfs", line 11, in <module>
    load_entry_point('aws-adfs==0.3.0', 'console_scripts', 'aws-adfs')()
  File "/Library/Python/2.7/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/Library/Python/2.7/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/Library/Python/2.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Library/Python/2.7/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Library/Python/2.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/Library/Python/2.7/site-packages/aws_adfs/login.py", line 82, in login
    principal_roles, assertion, aws_session_duration = authenticator.authenticate(config)
  File "/Library/Python/2.7/site-packages/aws_adfs/authenticator.py", line 16, in authenticate
    password=password,
  File "/Library/Python/2.7/site-packages/aws_adfs/html_roles_fetcher.py", line 66, in fetch_html_encoded_roles
    '''.format(authentication_url, response.request.headers, response.status_code, response.headers, response.text))
UnicodeEncodeError: 'ascii' codec can't encode character u'\u201c' in position 18498: ordinal not in range(128)
```

**Comment by venth** [comment]
Unfortunately I not able to reproduce the issue. I think that setup utf-8 as default encoding for stdout stream used in logging is sufficient. The change is already in master. @anthony-estevez-zefr, may I ask you to verify, if the introduced change corrected UnicodeEncodeError issue?

**Comment by anthony-estevez-zefr** [comment]
It appears that it did not. Is there anything else I can provide that may shred some light on this issue?

2017-04-03 10:48:10,914 [html_roles_fetcher html_roles_fetcher.py:fetch_html_encoded_roles] [23773-MainProcess] [140596703614720-MainThread] - DEBUG: Attempt to load authentication cookies into session failed. Re-authentication will be performed. The error: 
2017-04-03 10:48:10,915 [connectionpool connectionpool.py:_new_conn] [23773-MainProcess] [140596703614720-MainThread] - DEBUG: Starting new HTTPS connection (1): SITE
2017-04-03 10:48:11,381 [connectionpool connectionpool.py:_make_request] [23773-MainProcess] [140596703614720-MainThread] - DEBUG: https://SITE:443 "POST /adfs/ls/IdpInitiatedSignOn.aspx?loginToRp=urn:amazon:webservices HTTP/1.1" 200 19245
Traceback (most recent call last):
  File "/usr/local/bin/aws-adfs", line 11, in <module>
    load_entry_point('aws-adfs==0.3.1', 'console_scripts', 'aws-adfs')()
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "build/bdist.linux-x86_64/egg/aws_adfs/login.py", line 82, in login
  File "build/bdist.linux-x86_64/egg/aws_adfs/authenticator.py", line 16, in authenticate
  File "build/bdist.linux-x86_64/egg/aws_adfs/html_roles_fetcher.py", line 66, in fetch_html_encoded_roles
UnicodeEncodeError: 'ascii' codec can't encode character u'\u201c' in position 18498: ordinal not in range(128)

**Comment by venth** [comment]
Did you install from latest master commit by cloning repo locally and performing pip install -e. ?

**Comment by anthony-estevez-zefr** [comment]
Yes, I just did and it give me the following error.
```
2017-04-03 11:45:36,095 [connectionpool connectionpool.py:_new_conn] [14822-MainProcess] [140736347890624-MainThread] - DEBUG: Starting new HTTPS connection (1): SITE
2017-04-03 11:45:36,932 [connectionpool connectionpool.py:_make_request] [14822-MainProcess] [140736347890624-MainThread] - DEBUG: SITE:443 "POST /adfs/ls/IdpInitiatedSignOn.aspx?loginToRp=urn:amazon:webservices HTTP/1.1" 200 19245
Traceback (most recent call last):
  File "/usr/local/bin/aws-adfs", line 11, in <module>
    load_entry_point('aws-adfs', 'console_scripts', 'aws-adfs')()
  File "/Library/Python/2.7/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/Library/Python/2.7/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/Library/Python/2.7/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Library/Python/2.7/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Library/Python/2.7/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/git/aws-adfs/aws_adfs/login.py", line 82, in login
    principal_roles, assertion, aws_session_duration = authenticator.authenticate(config)
  File "/git/aws-adfs/aws_adfs/authenticator.py", line 16, in authenticate
    password=password,
  File "/git/aws-adfs/aws_adfs/html_roles_fetcher.py", line 66, in fetch_html_encoded_roles
    '''.format(authentication_url, response.request.headers, response.status_code, response.headers, response.text))
UnicodeEncodeError: 'ascii' codec can't encode character u'\u201c' in position 18498: ordinal not in range(128)
```

**Comment by anthony-estevez-zefr** [comment]
Does the application account for redirects? Our site sso.zefr.com redirects to sso.zefr.com/adfs/ls/IdpInitiatedSignOn.aspx

**Comment by venth** [comment]
Thanks for the hint. I'll check it Today.

**Comment by anthony-estevez-zefr** [comment]
I found that when I changed the python version to 3 and I don't run into this unicode issue.

**Comment by venth** [comment]
Fix delivered with version 0.3.1

**Comment by anthony-estevez-zefr** [comment]
Fantastic, I just tested it.
